internal/provider/resource_tfe_test_variable_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccTFETestVariable_basic(t *testing.T) {
+	variable := &tfe.Variable{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		CheckDestroy:             testAccCheckTFETestVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFETestVariable_test_variable(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFETestVariableExists(
+						"tfe_test_variable.foobar", variable),
+					testAccCheckTFEVariableAttributes(variable),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "key", "key_test"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "value", "value_test"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "description", "some description"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "category", "env"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "hcl", "false"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "sensitive", "false"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTFETestVariable_update(t *testing.T) {
+	variable := &tfe.Variable{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		CheckDestroy:             testAccCheckTFETestVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFETestVariable_test_variable(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFETestVariableExists(
+						"tfe_test_variable.foobar", variable),
+					testAccCheckTFEVariableAttributes(variable),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "key", "key_test"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "value", "value_test"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "description", "some description"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "category", "env"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "hcl", "false"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "sensitive", "false"),
+				),
+			},
+
+			{
+				Config: testAccTFETestVariable_update(rInt),
```

Comment:
😅 Please excuse my ignorance of the provider test setup. Is this supposed to update a specific variable, because it seems like the config provided removes one and creates another?

Reply:
This does update the particular variable. The initial basic test called on line 60 creates the variable and then the terraform code block called on line 81 updates that same variable. I've verified this via a debugger as well.

Reply:
Talked over zoom: it's a misunderstanding on my part, I glossed over the TF configuration. I'm convinced the tests correctly test the behaviour. 👍🏻